src/analysis/allman.d

```diff
@@ -0,0 +1,203 @@
+// Distributed under the Boost Software License, Version 1.0.
+//    (See accompanying file LICENSE_1_0.txt or copy at
+//          http://www.boost.org/LICENSE_1_0.txt)
+
+module analysis.allman;
+
+import dparse.lexer;
+import dparse.ast;
+import analysis.base : BaseAnalyzer;
+import dsymbol.scope_ : Scope;
+
+/**
+Checks for the allman style (braces should be on their own line)
+
+------------
+if (param < 0) {
+
+}
+------------
+
+should be
+
+------------
+if (param < 0)
+{
+
+}
+------------
+*/
+class AllManCheck : BaseAnalyzer
+{
+    ubyte[] code;
+
+    ///
+    this(string fileName, ubyte[] code, bool skipTests = false)
```

Comment:
I didn't know how I could access the tokens based on a statement and look ahead/before. I only found `startLocation`